File: src/main/java/top/axbt/keshe/service/CompanyinfoService.java
package top.axbt.keshe.service;

import top.axbt.entity.PageResult;
import top.axbt.keshe.domain.TbCompanyinfo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CompanyinfoService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbCompanyinfo> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbCompanyinfo companyinfo);
	
	
	/**
	 * 修改
	 */
	public void update(TbCompanyinfo companyinfo);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCompanyinfo findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbCompanyinfo companyinfo, int pageNum, int pageSize);
	
}

File: src/main/webapp/js/service/companyinfoService.js
//服务层
app.service('companyinfoService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../companyinfo/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../companyinfo/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../companyinfo/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../companyinfo/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../companyinfo/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../companyinfo/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../companyinfo/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});
